providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/features/AWSSecurityGroupApiMockTest.java

```diff
@@ -0,0 +1,270 @@
+         "      <ipPermissionsEgress>",
+         "        <item>",
+         "          <ipProtocol>-1</ipProtocol>",
+         "          <groups/>",
+         "          <ipRanges>",
+         "            <item>",
+         "              <cidrIp>0.0.0.0/0</cidrIp>",
+         "            </item>",
+         "          </ipRanges>",
+         "          <prefixListIds/>",
+         "        </item>",
+         "      </ipPermissionsEgress>",
+         "    </item>",
+         "  </securityGroupInfo>",
+         "</DescribeSecurityGroupsResponse>");
+
+   private final String createSecurityGroupResponse = Joiner.on("\n").join(
+         "<CreateSecurityGroupResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-11-15/\">",
+         "  <requestId>59dbff89-35bd-4eac-99ed-be587EXAMPLE</requestId>",
+         "  <return>true</return>",
+         "  <groupId>sg-0a42d66a</groupId>",
+         "</CreateSecurityGroupResponse>");
+
+   private final String authorizeSecurityGroupIngressResponse = Joiner.on("\n").join(
+            "<AuthorizeSecurityGroupIngressResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-11-15/\">",
+            "  <requestId>59dbff89-35bd-4eac-99ed-be587EXAMPLE</requestId>",
+            "  <return>true</return>",
+            "</AuthorizeSecurityGroupIngressResponse>");
+
+   private final String revokeSecurityGroupIngressResponse = Joiner.on("\n").join(
+         "<RevokeSecurityGroupIngressResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-11-15/\">",
+         "  <requestId>59dbff89-35bd-4eac-99ed-be587EXAMPLE</requestId>",
+         "  <return>true</return>",
+         "</RevokeSecurityGroupIngressResponse>");
+
+   private final String deleteSecurityGroupResponse = Joiner.on("\n").join(
+         "<DeleteSecurityGroupResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-11-15/\">",
+         "  <requestId>59dbff89-35bd-4eac-99ed-be587EXAMPLE</requestId>",
+         "  <return>true</return>",
+         "</DeleteSecurityGroupResponse>");
+
+   @SuppressWarnings("deprecation")
+   public void describeSecurityGroups() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION, new MockResponse().setBody(describeSecurityGroupsResponse));
+
+      Set<SecurityGroup> results = securityGroupApi().describeSecurityGroupsInRegion(DEFAULT_REGION);
+      SecurityGroup result = Iterables.getOnlyElement(results);
+      assertEquals(result.getId(), "sg-1a2b3c4d");
+      assertEquals(result.getRegion(), "us-east-1");
+      assertEquals(result.getName(), "WebServers");
+      assertEquals(result.getOwnerId(), "123456789012");
+      assertEquals(result.getDescription(), "Web Servers");
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DescribeSecurityGroups");
+   }
+
+   public void describeSecurityGroupsGiving404() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION, new MockResponse().setResponseCode(404));
```

Comment:
Ah, thanks for the pointer @andreaturli - I didn't know about that.

I'll leave it as-is here if that's ok with you. The response strings are defined in the class `AWSSecurityGroupApiMockTest`, rather than external resource files (following the pattern you used in your commit for `AWSSubnetApiMockTest`). Therefore `enqueueXml` wouldn't work unless we moved each response into its own resource file.